tests/test_pipelines.py

```diff
@@ -1876,6 +1924,53 @@ def test_stable_diffusion_img2img_pipeline(self):
         # img2img is flaky across GPUs even in fp32, so using MAE here
         assert np.abs(expected_image - image).mean() < 1e-2
 
+    @slow
+    @unittest.skipIf(torch_device == "cpu", "Stable diffusion is supposed to run on GPU")
+    def test_stable_diffusion_cycle_diffusion_pipeline(self):
+        init_image = load_image(
+            "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main"
+            "/cycle-diffusion/black_colored_car.png"
+        )
+        expected_image = load_image(
+            "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main"
+            "/img2img/fantasy_landscape.png"
+        )
+        init_image = init_image.resize((512, 512))
+        expected_image = np.array(expected_image, dtype=np.float32) / 255.0
+
+        model_id = "CompVis/stable-diffusion-v1-4"
+        scheduler = DDIMScheduler.from_config(model_id, subfolder="scheduler")
+
+        pipe = StableDiffusionImg2ImgPipeline.from_pretrained(
+            model_id,
+            scheduler=scheduler,
+            safety_checker=None
+        )
+        pipe.to(torch_device)
+        pipe.set_progress_bar_config(disable=None)
+        pipe.enable_attention_slicing()
+
+        source_prompt = "A black colored car"
+        prompt = "A red colored car"
+
+        generator = torch.Generator(device=torch_device).manual_seed(0)
+        output = pipe(
+             prompt=prompt,
+             source_prompt=source_prompt,
+             generator=generator,
+             init_image=init_image,
+             num_inference_steps=100,
+             eta=0.1,
+             strength=0.8,
+             guidance_scale=3,
+             source_guidance_scale=1,
+        )
+        image = output.images[0]
+
+
+        # image.save(...) doesn't yet produce super good results - not sure why
```

Comment:
This test currently doesn't work as expected I think . The resulting image is not really a red car. @ChenWu98 any idea what could be the problem here?

Reply:
It may need some parameter tweaks. Let me verify this.

Reply:
Just tried this example and it works with "blue car" prompt.